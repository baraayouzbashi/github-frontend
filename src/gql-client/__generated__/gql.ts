/* eslint-disable */
import * as types from './graphql';
import { TypedDocumentNode as DocumentNode } from '@graphql-typed-document-node/core';

/**
 * Map of all GraphQL operations in the project.
 *
 * This map has several performance disadvantages:
 * 1. It is not tree-shakeable, so it will include all operations in the project.
 * 2. It is not minifiable, so the string of a GraphQL query will be multiple times inside the bundle.
 * 3. It does not support dead code elimination, so it will add unused operations.
 *
 * Therefore it is highly recommended to use the babel or swc plugin for production.
 * Learn more about it here: https://the-guild.dev/graphql/codegen/plugins/presets/preset-client#reducing-bundle-size
 */
const documents = {
    "\n  query GetIssueWithCommentsByID($issueId: ID!, $numComments: Int!) {\n    node(id: $issueId) {\n      ... on Issue {\n        title\n        body\n        url\n        createdAt\n        state\n        author {\n          login\n        }\n        comments(first: $numComments) {\n          nodes {\n            body\n            createdAt\n            author {\n              login\n            }\n          }\n          pageInfo {\n            hasNextPage\n            endCursor\n          }\n        }\n      }\n    }\n  }\n": types.GetIssueWithCommentsByIdDocument,
    "\nquery GetRepositoryIssues($owner: String!, $name: String!, $states: [IssueState!], $IssuePagination: Int!) {\n  repository(owner: $owner, name: $name) {\n    issues(\n      states: $states\n      first: $IssuePagination\n      orderBy: {direction: DESC, field: CREATED_AT}\n    ) {\n      totalCount\n      nodes {\n        state\n        title\n        labels(first: 3) {\n          nodes {\n            color\n            name\n          }\n        }\n        comments {\n          totalCount\n        }\n        id\n      }\n    }\n  }\n}\n": types.GetRepositoryIssuesDocument,
    "\n  query SearchIssuesByTitleOrBody($query: String!, $numResults: Int!) {\n  search(query: $query, type: ISSUE, first: $numResults) {\n    edges {\n      node {\n        ... on Issue {\n          state\n          title\n          labels(first: 3) {\n            nodes {\n              color\n              name\n            }\n          }\n          comments {\n            totalCount\n          }\n          id\n          author {\n            login\n          }\n        }\n      }\n    }\n  }\n}\n": types.SearchIssuesByTitleOrBodyDocument,
};

/**
 * The gql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 *
 *
 * @example
 * ```ts
 * const query = gql(`query GetUser($id: ID!) { user(id: $id) { name } }`);
 * ```
 *
 * The query argument is unknown!
 * Please regenerate the types.
 */
export function gql(source: string): unknown;

/**
 * The gql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function gql(source: "\n  query GetIssueWithCommentsByID($issueId: ID!, $numComments: Int!) {\n    node(id: $issueId) {\n      ... on Issue {\n        title\n        body\n        url\n        createdAt\n        state\n        author {\n          login\n        }\n        comments(first: $numComments) {\n          nodes {\n            body\n            createdAt\n            author {\n              login\n            }\n          }\n          pageInfo {\n            hasNextPage\n            endCursor\n          }\n        }\n      }\n    }\n  }\n"): (typeof documents)["\n  query GetIssueWithCommentsByID($issueId: ID!, $numComments: Int!) {\n    node(id: $issueId) {\n      ... on Issue {\n        title\n        body\n        url\n        createdAt\n        state\n        author {\n          login\n        }\n        comments(first: $numComments) {\n          nodes {\n            body\n            createdAt\n            author {\n              login\n            }\n          }\n          pageInfo {\n            hasNextPage\n            endCursor\n          }\n        }\n      }\n    }\n  }\n"];
/**
 * The gql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function gql(source: "\nquery GetRepositoryIssues($owner: String!, $name: String!, $states: [IssueState!], $IssuePagination: Int!) {\n  repository(owner: $owner, name: $name) {\n    issues(\n      states: $states\n      first: $IssuePagination\n      orderBy: {direction: DESC, field: CREATED_AT}\n    ) {\n      totalCount\n      nodes {\n        state\n        title\n        labels(first: 3) {\n          nodes {\n            color\n            name\n          }\n        }\n        comments {\n          totalCount\n        }\n        id\n      }\n    }\n  }\n}\n"): (typeof documents)["\nquery GetRepositoryIssues($owner: String!, $name: String!, $states: [IssueState!], $IssuePagination: Int!) {\n  repository(owner: $owner, name: $name) {\n    issues(\n      states: $states\n      first: $IssuePagination\n      orderBy: {direction: DESC, field: CREATED_AT}\n    ) {\n      totalCount\n      nodes {\n        state\n        title\n        labels(first: 3) {\n          nodes {\n            color\n            name\n          }\n        }\n        comments {\n          totalCount\n        }\n        id\n      }\n    }\n  }\n}\n"];
/**
 * The gql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function gql(source: "\n  query SearchIssuesByTitleOrBody($query: String!, $numResults: Int!) {\n  search(query: $query, type: ISSUE, first: $numResults) {\n    edges {\n      node {\n        ... on Issue {\n          state\n          title\n          labels(first: 3) {\n            nodes {\n              color\n              name\n            }\n          }\n          comments {\n            totalCount\n          }\n          id\n          author {\n            login\n          }\n        }\n      }\n    }\n  }\n}\n"): (typeof documents)["\n  query SearchIssuesByTitleOrBody($query: String!, $numResults: Int!) {\n  search(query: $query, type: ISSUE, first: $numResults) {\n    edges {\n      node {\n        ... on Issue {\n          state\n          title\n          labels(first: 3) {\n            nodes {\n              color\n              name\n            }\n          }\n          comments {\n            totalCount\n          }\n          id\n          author {\n            login\n          }\n        }\n      }\n    }\n  }\n}\n"];

export function gql(source: string) {
  return (documents as any)[source] ?? {};
}

export type DocumentType<TDocumentNode extends DocumentNode<any, any>> = TDocumentNode extends DocumentNode<  infer TType,  any>  ? TType  : never;