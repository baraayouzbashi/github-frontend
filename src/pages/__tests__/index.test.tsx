import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Home from "../index";
import { GetRepositoryIssuesQuery } from "@/gql-client/__generated__/graphql";

const data = {
  repository: {
    name: "name of repo",
    description: "description of the repo",
    issues: {
      nodes: [],
      totalCount: 0,
    },
  },
} as GetRepositoryIssuesQuery;

describe("Home", () => {
  it("renders a heading", () => {
    render(<Home data={data} />);
    const heading = screen.queryByRole("heading", { level: 1 });
    expect(heading).toBeInTheDocument();
    expect(heading.innerHTML).toEqual(data.repository?.name);
  });
  it("renders Loading screen", () => {
    render(<Home data={{ repository: null }} />);
    const loading = screen.getByText(/Loading/);
    expect(loading).toBeTruthy();
  });
  it("renders homepage unchanged", () => {
    const { container } = render(<Home data={data} />);
    expect(container).toMatchSnapshot();
  });
});
