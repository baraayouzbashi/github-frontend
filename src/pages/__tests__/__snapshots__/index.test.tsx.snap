// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
.c0 {
  max-width: 800px;
  margin: 0 auto;
  padding: 2rem;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
}

.c1 {
  font-weight: bold;
  font-size: 92px;
  color: var(--color-secondary);
  margin-bottom: 1rem;
}

.c2 {
  font-size: 1.1rem;
  margin-bottom: 2rem;
  line-height: 1.5;
}

.c3 {
  width: 100%;
  padding: 0.75rem;
  font-size: 1rem;
  border: 1px solid var(--color-primary);
  border-radius: 4px;
  margin-bottom: 1.5rem;
  box-shadow: 0 2px 4px rgba(0,0,0,0.1);
  transition: box-shadow 0.3s ease-in-out;
}

.c3:focus {
  outline: none;
  box-shadow: 0 0 0 2px var(--color-accent);
}

.c4 {
  display: flex;
  flex-direction: column;
  gap: 1.5rem;
  list-style-type: none;
}

<div>
  <div
    class="c0"
  >
    <h1
      class="c1"
    >
      name of repo
    </h1>
    <p
      class="c2"
    >
      description of the repo
    </p>
    <input
      aria-label="Search issues"
      class="c3"
      placeholder="Search issues..."
      type="text"
      value=""
    />
    <ul
      class="c4"
    />
  </div>
</div>
`;
